### PR TITLE
smbus: fix invalid memory access in read_word()

### PR DESCRIPTION
**Pull request by bkueng**
read_word() expected 3 bytes (uint16_t + PEC byte), but was passed an address to an uint16_t value.

write_word() is changed to be type-safe as well.

@dakejahl can you test this?

### CODE REVIEW

**Comment by dakejahl** [comment]
Since I am no longer at Teal I do not have the hardware.

**Comment by mcsauder** [comment]
Hi @AlexKlimaj, do you happen to have alternative hardware you could test with?

**Comment by AlexKlimaj** [comment]
Unfortunately I do not. @dakejahl has a fuel gauge in the same TI line as the BQ40Z50-R2 that may work. Someone at Teal will have to test this.

**Comment by dakejahl** [comment]
Yes, when I get that board up and running I can at the very least test the
`smbus` library functions. That may be a little while still.

**Comment by bkueng** [comment]
Thanks for the quick responses.

I added several more fixes.
@themcfadden I see you are the maintainer for the Teal now. Can you test this please?

**Comment by themcfadden** [comment]
I'll spend some time on it this week.

**Comment by themcfadden** [comment]
@bkueng driver seems to be working well. I haven't tested the write command. I don't have an easy way to do it.
status command does not show up in help.

File: src/drivers/batt_smbus/batt_smbus.cpp [hunk 1]
```diff
@@ -132,22 +117,23 @@ int BATT_SMBUS::task_spawn(int argc, char *argv[])
...
-			_task_id = task_id_is_work_queue;
+			SMBus *interface = new SMBus(smbus_bus_options[i].busnum, BATT_SMBUS_ADDR);
+			BATT_SMBUS *dev = new BATT_SMBUS(interface, smbus_bus_options[i].devpath);
 
 			int result = dev->get_startup_info();
 
 			if (result != PX4_OK) {
-				return PX4_ERROR;
+				delete dev;
+				continue;
```
Comment (dagar) [inline src/drivers/batt_smbus/batt_smbus.cpp]: Why is this continuing?

**Comment by bkueng** [comment]
> status command does not show up in help.

I added it as well. Thanks for testing.

**Pull request merged**